### ISSUE TITLE
Docker containers not removed by image name alone 

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29909]
From @metajungle on 2016-02-03T17:45:28Z
##### ISSUE TYPE

Bug Report
##### COMPONENT NAME

_docker module
##### ANSIBLE VERSION

N/A
##### SUMMARY

The following command does not remove running containers started from a given image, which would be nice to have: 

```
$ ansible -i prod docker-hosts -s -m docker -a "image=repo/image-name:latest state=absent"
10.12.6.31 | SUCCESS => {
    "ansible_facts": {
        "docker_containers": []
    }, 
    "changed": false, 
    "msg": "No action taken.", 
    "reload_reasons": null, 
    "summary": {
        "created": 0, 
        "killed": 0, 
        "pulled": 0, 
        "removed": 0, 
        "restarted": 0, 
        "started": 0, 
        "stopped": 0
    }
}
```

However, if I add the name of the container, it is stopped and removed: 

```
$ ansible -i prod docker-hosts -s -m docker -a "image=repo/image-name:latest name=name_of_container state=absent"
10.12.6.31 | SUCCESS => {
    ...
    "changed": true, 
    "msg": "removed 1 container, stopped 1 container.", 
    "reload_reasons": null, 
    "summary": {
        "created": 0, 
        "killed": 0, 
        "pulled": 0, 
        "removed": 1, 
        "restarted": 0, 
        "started": 0, 
        "stopped": 1
    }
}
```

It would be nice to be able to stop containers based on the images they are launched from, without having to explicitly specify the name. 

Copied from original issue: ansible/ansible-modules-core#2948

**Comment by ansibot** [comment]
From @ansibot on 2016-02-03T17:45:28Z



@cove, @softzilla, @zfil, @ThomasSteinbach, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @dusdanig @joshuaconner
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gundalow** [comment]
The `cloud/docker/_docker.py` no longer exists in Ansible 2.5.0.

For information about the newer Docker modules to use, please see http://docs.ansible.com/ansible/devel/scenario_guides/guide_docker.html

If after reviewing the newer modules you believe this issue/PR would still be useful addition to the newer modules, please create a *new* issue/PR against the new module and reference this issue/PR in the comments.

Closing this. Any comments on here will be ignored.
